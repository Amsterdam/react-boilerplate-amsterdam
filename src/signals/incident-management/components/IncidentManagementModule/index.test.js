import React from 'react';
import { shallow } from 'enzyme';

import { IncidentManagementModule } from './index';

describe('<IncidentManagementModule />', () => {
  let props;

  beforeEach(() => {
    props = {
      match: { params: { id: 1 }, url: 'http://test/url' },
      isAuthenticated: true
    };
  });

  it('should render correctly when authenticated', () => {
    const renderedComponent = shallow(
      <IncidentManagementModule {...props} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render correctly when not authenticated', () => {
    props.isAuthenticated = false;
    const renderedComponent = shallow(
      <IncidentManagementModule {...props} />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
