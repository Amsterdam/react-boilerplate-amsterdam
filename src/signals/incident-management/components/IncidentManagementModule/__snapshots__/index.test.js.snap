// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncidentManagementModule /> should render correctly when authenticated 1`] = `
<div
  className="manage-incident"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <Route
        exact={true}
        path="http://test/url/incidents"
        render={[Function]}
      />
      <Route
        exact={true}
        path="http://test/url/incident/:id"
        render={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`<IncidentManagementModule /> should render correctly when not authenticated on management page 1`] = `
<div
  className="manage-incident"
>
  <Route
    component={[Function]}
  />
</div>
`;

exports[`<IncidentManagementModule /> should render correctly when not authenticated on public page 1`] = `
<div
  className="manage-incident"
>
  <Route
    component={[Function]}
  />
</div>
`;
